CDP4ShellDialogs/ViewModels/PluginRowViewModel.cs

```diff
@@ -42,26 +45,60 @@ public class PluginRowViewModel : ReactiveObject
         /// </summary>
         private string company;
 
+        /// <summary>
+        /// The backing field for the <see cref="IsPluginEnabled"/> property
+        /// </summary>
+        private bool isPluginEnabled;
+
+        /// <summary>
+        /// The backing field for the <see cref="IsRowEnabled"/> property
+        /// </summary>
+        private bool isRowEnabled;
+
+        /// <summary>
+        /// The backing field for the <see cref="IsMandatory"/> property
+        /// </summary>
+        private bool isMandatory;
+
+        /// <summary>
+        /// The backing field for the <see cref="IsRowDirty"/> property
+        /// </summary>
+        private bool isRowDirty;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="PluginRowViewModel"/> class.
         /// </summary>
         /// <param name="module">
         /// The <see cref="IModule"/> to extract information from.
         /// </param>
-        public PluginRowViewModel(IModule module)
+        public PluginRowViewModel(PluginSettingsMetaData module)
+        {
+            this.key = module.Key;
+            this.assemblyName = module.Assembly;
+            this.name = module.Name;
+            this.description = module.Description;
+            this.version = module.Version;
+            this.company = module.Company;
+            this.isPluginEnabled = module.IsEnabled;
+            this.isRowEnabled = !module.IsMandatory;
+            this.isMandatory = module.IsMandatory;
+            this.PropertyChanged += (s, e) => { this.IsRowDirty = true; };
```

Comment:
I thought this was not necessary anymore???

Reply:
I am not sure of this. What is not necessary?